CHANGELOG: Update release date for v3.5.15

jmhbnz opened the pull request:
We have now published v3.5.15, let's ensure the release date is added to our `CHANGELOG`.

ivanvc commented:
It looks good, but should we also add the next version (3.5.16) with TBD at the top of the file? The last time I updated, I added it because I saw previous PRs had it.

jmhbnz commented:
> It looks good, but should we also add the next version (3.5.16) with TBD at the top of the file? The last time I updated, I added it because I saw previous PRs had it.

Good call, updated!

ivanvc reviewed (approved):
/lgtm Thanks, @jmhbnz

jmhbnz commented:
/retest
